lib/internal/Magento/Framework/App/Router/NoRouteHandler.php

```diff
@@ -39,9 +41,9 @@ public function process(\Magento\Framework\App\RequestInterface $request)
             $noRoute = [];
         }
 
-        $moduleName = isset($noRoute[0]) ? $noRoute[0] : 'core';
-        $actionPath = isset($noRoute[1]) ? $noRoute[1] : 'index';
-        $actionName = isset($noRoute[2]) ? $noRoute[2] : 'index';
+        $moduleName = $noRoute[0] ?? 'cms';
```

Comment:
Didn't it change fallback order now?

Please avoid FQCN changes not related to bugfix.

Reply:
Vlad, hi! All changes were reverted, I only left description for classes for static tests passing.